diff --git a/pcdet/__init__.py b/pcdet/__init__.py
index 9fdf7d2..d30ef7b 100644
--- a/pcdet/__init__.py
+++ b/pcdet/__init__.py
@@ -12,7 +12,7 @@ def get_git_commit_number():
     if not (Path(__file__).parent / '../.git').exists():
         return '0000000'
 
-    cmd_out = subprocess.run(['git', 'rev-parse', 'HEAD'], stdout=subprocess.PIPE)
+    cmd_out = subprocess.run(['git', 'rev-parse', 'HEAD'], stdout=subprocess.PIPE, cwd=Path(__file__).resolve().parent.parent)
     git_commit_number = cmd_out.stdout.decode('utf-8')[:7]
     return git_commit_number
 
diff --git a/pcdet/models/backbones_2d/base_bev_backbone.py b/pcdet/models/backbones_2d/base_bev_backbone.py
index 07fe70f..65a3001 100644
--- a/pcdet/models/backbones_2d/base_bev_backbone.py
+++ b/pcdet/models/backbones_2d/base_bev_backbone.py
@@ -57,7 +57,7 @@ class BaseBEVBackbone(nn.Module):
                         nn.ReLU()
                     ))
                 else:
-                    stride = np.round(1 / stride).astype(np.int)
+                    stride = np.round(1 / stride).astype(int)
                     self.deblocks.append(nn.Sequential(
                         nn.Conv2d(
                             num_filters[idx], num_upsample_filters[idx],
diff --git a/pcdet/ops/pointnet2/pointnet2_batch/src/ball_query.cpp b/pcdet/ops/pointnet2/pointnet2_batch/src/ball_query.cpp
index 67784c6..925553c 100644
--- a/pcdet/ops/pointnet2/pointnet2_batch/src/ball_query.cpp
+++ b/pcdet/ops/pointnet2/pointnet2_batch/src/ball_query.cpp
@@ -7,12 +7,10 @@ All Rights Reserved 2018.
 
 #include <torch/serialize/tensor.h>
 #include <vector>
-#include <THC/THC.h>
 #include <cuda.h>
 #include <cuda_runtime_api.h>
 #include "ball_query_gpu.h"
 
-extern THCState *state;
 
 #define CHECK_CUDA(x) do { \
 	  if (!x.type().is_cuda()) { \
diff --git a/pcdet/ops/pointnet2/pointnet2_batch/src/group_points.cpp b/pcdet/ops/pointnet2/pointnet2_batch/src/group_points.cpp
index 5565700..5dd8978 100644
--- a/pcdet/ops/pointnet2/pointnet2_batch/src/group_points.cpp
+++ b/pcdet/ops/pointnet2/pointnet2_batch/src/group_points.cpp
@@ -9,10 +9,8 @@ All Rights Reserved 2018.
 #include <cuda.h>
 #include <cuda_runtime_api.h>
 #include <vector>
-#include <THC/THC.h>
 #include "group_points_gpu.h"
 
-extern THCState *state;
 
 
 int group_points_grad_wrapper_fast(int b, int c, int n, int npoints, int nsample, 
diff --git a/pcdet/ops/pointnet2/pointnet2_batch/src/interpolate.cpp b/pcdet/ops/pointnet2/pointnet2_batch/src/interpolate.cpp
index c2496e8..cea8375 100644
--- a/pcdet/ops/pointnet2/pointnet2_batch/src/interpolate.cpp
+++ b/pcdet/ops/pointnet2/pointnet2_batch/src/interpolate.cpp
@@ -7,7 +7,6 @@ All Rights Reserved 2018.
 
 #include <torch/serialize/tensor.h>
 #include <vector>
-#include <THC/THC.h>
 #include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -15,7 +14,6 @@ All Rights Reserved 2018.
 #include <cuda_runtime_api.h>
 #include "interpolate_gpu.h"
 
-extern THCState *state;
 
 
 void three_nn_wrapper_fast(int b, int n, int m, at::Tensor unknown_tensor, 
diff --git a/pcdet/ops/pointnet2/pointnet2_batch/src/sampling.cpp b/pcdet/ops/pointnet2/pointnet2_batch/src/sampling.cpp
index ee65986..85f5871 100644
--- a/pcdet/ops/pointnet2/pointnet2_batch/src/sampling.cpp
+++ b/pcdet/ops/pointnet2/pointnet2_batch/src/sampling.cpp
@@ -8,11 +8,9 @@ All Rights Reserved 2018.
 #include <torch/serialize/tensor.h>
 #include <ATen/cuda/CUDAContext.h>
 #include <vector>
-#include <THC/THC.h>
 
 #include "sampling_gpu.h"
 
-extern THCState *state;
 
 
 int gather_points_wrapper_fast(int b, int c, int n, int npoints, 
diff --git a/pcdet/ops/pointnet2/pointnet2_stack/src/ball_query.cpp b/pcdet/ops/pointnet2/pointnet2_stack/src/ball_query.cpp
index 8ae8f23..afd7d2e 100644
--- a/pcdet/ops/pointnet2/pointnet2_stack/src/ball_query.cpp
+++ b/pcdet/ops/pointnet2/pointnet2_stack/src/ball_query.cpp
@@ -7,12 +7,10 @@ All Rights Reserved 2019-2020.
 
 #include <torch/serialize/tensor.h>
 #include <vector>
-#include <THC/THC.h>
 #include <cuda.h>
 #include <cuda_runtime_api.h>
 #include "ball_query_gpu.h"
 
-extern THCState *state;
 
 #define CHECK_CUDA(x) do { \
   if (!x.type().is_cuda()) { \
diff --git a/pcdet/ops/pointnet2/pointnet2_stack/src/group_points.cpp b/pcdet/ops/pointnet2/pointnet2_stack/src/group_points.cpp
index 4f48b6d..d882c59 100644
--- a/pcdet/ops/pointnet2/pointnet2_stack/src/group_points.cpp
+++ b/pcdet/ops/pointnet2/pointnet2_stack/src/group_points.cpp
@@ -9,10 +9,8 @@ All Rights Reserved 2019-2020.
 #include <cuda.h>
 #include <cuda_runtime_api.h>
 #include <vector>
-#include <THC/THC.h>
 #include "group_points_gpu.h"
 
-extern THCState *state;
 #define CHECK_CUDA(x) do { \
   if (!x.type().is_cuda()) { \
     fprintf(stderr, "%s must be CUDA tensor at %s:%d\n", #x, __FILE__, __LINE__); \
diff --git a/pcdet/ops/pointnet2/pointnet2_stack/src/interpolate.cpp b/pcdet/ops/pointnet2/pointnet2_stack/src/interpolate.cpp
index 76135ee..6ca94b7 100644
--- a/pcdet/ops/pointnet2/pointnet2_stack/src/interpolate.cpp
+++ b/pcdet/ops/pointnet2/pointnet2_stack/src/interpolate.cpp
@@ -7,7 +7,6 @@ All Rights Reserved 2019-2020.
 
 #include <torch/serialize/tensor.h>
 #include <vector>
-#include <THC/THC.h>
 #include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -15,7 +14,6 @@ All Rights Reserved 2019-2020.
 #include <cuda_runtime_api.h>
 #include "interpolate_gpu.h"
 
-extern THCState *state;
 
 #define CHECK_CUDA(x) do { \
   if (!x.type().is_cuda()) { \
diff --git a/pcdet/ops/pointnet2/pointnet2_stack/src/sampling.cpp b/pcdet/ops/pointnet2/pointnet2_stack/src/sampling.cpp
index 7cab9c2..8cb56c7 100644
--- a/pcdet/ops/pointnet2/pointnet2_stack/src/sampling.cpp
+++ b/pcdet/ops/pointnet2/pointnet2_stack/src/sampling.cpp
@@ -1,11 +1,9 @@
 #include <torch/serialize/tensor.h>
 #include <ATen/cuda/CUDAContext.h>
 #include <vector>
-#include <THC/THC.h>
 
 #include "sampling_gpu.h"
 
-extern THCState *state;
 #define CHECK_CUDA(x) do { \
   if (!x.type().is_cuda()) { \
     fprintf(stderr, "%s must be CUDA tensor at %s:%d\n", #x, __FILE__, __LINE__); \
diff --git a/pcdet/ops/pointnet2/pointnet2_stack/src/vector_pool.cpp b/pcdet/ops/pointnet2/pointnet2_stack/src/vector_pool.cpp
index 308beea..a228227 100644
--- a/pcdet/ops/pointnet2/pointnet2_stack/src/vector_pool.cpp
+++ b/pcdet/ops/pointnet2/pointnet2_stack/src/vector_pool.cpp
@@ -10,12 +10,10 @@ All Rights Reserved 2020.
 
 #include <torch/serialize/tensor.h>
 #include <vector>
-#include <THC/THC.h>
 #include <cuda.h>
 #include <cuda_runtime_api.h>
 #include "vector_pool_gpu.h"
 
-extern THCState *state;
 
 #define CHECK_CUDA(x) do { \
   if (!x.type().is_cuda()) { \
diff --git a/pcdet/ops/pointnet2/pointnet2_stack/src/voxel_query.cpp b/pcdet/ops/pointnet2/pointnet2_stack/src/voxel_query.cpp
index 735fe5e..5dc2b77 100644
--- a/pcdet/ops/pointnet2/pointnet2_stack/src/voxel_query.cpp
+++ b/pcdet/ops/pointnet2/pointnet2_stack/src/voxel_query.cpp
@@ -1,6 +1,5 @@
 #include <torch/serialize/tensor.h>
 #include <vector>
-#include <THC/THC.h>
 #include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -8,7 +7,6 @@
 #include <cuda_runtime_api.h>
 #include "voxel_query_gpu.h"
 
-extern THCState *state;
 
 #define CHECK_CUDA(x) do { \
   if (!x.type().is_cuda()) { \
